xbmc/interfaces/json-rpc/VideoLibrary.cpp

```diff
@@ -472,6 +472,47 @@ JSONRPC_STATUS CVideoLibrary::GetGenres(const std::string &method, ITransportLay
   return OK;
 }
 
+JSONRPC_STATUS CVideoLibrary::GetTags(const std::string &method, ITransportLayer *transport, IClient *client, const CVariant &parameterObject, CVariant &result)
+{
+  std::string media = parameterObject["type"].asString();
+  StringUtils::ToLower(media);
+  int idContent = -1;
+
+  std::string strPath = "videodb://";
+  /* select which video content to get tags from*/
+  if (media == MediaTypeMovie)
+  {
+    idContent = VIDEODB_CONTENT_MOVIES;
+    strPath += "movies";
+  }
+  else if (media == MediaTypeTvShow)
+  {
+    idContent = VIDEODB_CONTENT_TVSHOWS;
+    strPath += "tvshows";
+  }
+  else if (media == MediaTypeMusicVideo)
+  {
+    idContent = VIDEODB_CONTENT_MUSICVIDEOS;
+    strPath += "musicvideos";
+  }
+  strPath += "/tags/";
+
+  CVideoDatabase videodatabase;
+  if (!videodatabase.Open())
+    return InternalError;
+
+  CFileItemList items;
+  if (!videodatabase.GetTagsNav(strPath, items, idContent))
+    return InternalError;
+
+  /* need to set strTitle in each item*/
+  for (unsigned int i = 0; i < static_cast<unsigned int>(items.Size()); i++)
```

Comment:
This cast will be done on each iteration. I suggest something like `for (auto &item : items)`.